File: myFinance_improved_V2/lib/features/journal_input/domain/entities/transaction_line.dart
// Domain Entity: TransactionLine
// Pure business entity with no dependencies on frameworks

import 'package:freezed_annotation/freezed_annotation.dart';

part 'transaction_line.freezed.dart';

@freezed
class TransactionLine with _$TransactionLine {
  const factory TransactionLine({
    String? accountId,
    String? accountName,
    String? categoryTag,
    String? description,
    @Default(0.0) double amount,
    @Default(true) bool isDebit,
    String? counterpartyId,
    String? counterpartyName,
    String? counterpartyStoreId,
    String? counterpartyStoreName,
    String? cashLocationId,
    String? cashLocationName,
    String? cashLocationType,
    String? linkedCompanyId,
    String? counterpartyCashLocationId,

    // Debt related fields
    String? debtCategory,
    double? interestRate,
    DateTime? issueDate,
    DateTime? dueDate,
    String? debtDescription,

    // Fixed asset fields
    String? fixedAssetName,
    double? salvageValue,
    DateTime? acquisitionDate,
    int? usefulLife,

    // Account mapping fields for internal transactions
    Map<String, dynamic>? accountMapping,
  }) = _TransactionLine;
}

File: website/src/features/journal_input/domain/entities/TransactionLine.ts
/**
 * TransactionLine Entity
 * Represents a single transaction line (debit or credit) in a journal entry
 */

export class TransactionLine {
  constructor(
    public readonly isDebit: boolean,
    public readonly accountId: string,
    public readonly accountName: string,
    public readonly amount: number,
    public readonly description: string,
    public readonly categoryTag: string | null,
    public readonly cashLocationId: string | null,
    public readonly cashLocationName: string | null,
    public readonly cashLocationType: string | null,
    public readonly counterpartyId: string | null,
    public readonly counterpartyName: string | null,
    public readonly counterpartyStoreId: string | null,
    public readonly counterpartyStoreName: string | null,
    public readonly debtCategory: string | null,
    // Additional debt fields for p_lines alignment
    public readonly interestRate: number | null = null,
    public readonly interestAccountId: string | null = null,
    public readonly interestDueDay: number | null = null,
    public readonly issueDate: string | null = null,
    public readonly dueDate: string | null = null,
    public readonly debtDescription: string | null = null,
    public readonly linkedCompanyId: string | null = null,
    // Counterparty cash location for mirror journal (p_if_cash_location_id)
    public readonly counterpartyCashLocationId: string | null = null
  ) {}

  /**
   * Get transaction type display
   */
  get typeDisplay(): string {
    return this.isDebit ? 'DEBIT' : 'CREDIT';
  }

  /**
   * Check if transaction line is valid
   */
  get isValid(): boolean {
    return !!(this.accountId && this.amount > 0);
  }
}
